src/libraries/System.Net.Http/tests/UnitTests/Headers/HttpHeadersTest.cs

```diff
@@ -2359,6 +2359,13 @@ public void Add_LargeNumberOfHeaders_OperationsStillSupported(int numberOfHeader
             Assert.Equal(new[] { "newValue" }, valuesFor3);
         }
 
+        [Fact]
+        public void TryAddInvalidHeader_ShouldThrowFormatException()
+        {
+            MockHeaders headers = new MockHeaders();
+            AssertExtensions.Throws<FormatException>(() => headers.Remove("\u0080"), "The header name '\u0080' format is invalid.");
```

Comment:
This will fall if we ever allow to opt into localized exception messages. Better to just check the value is contained within the string.